3.16.1 does not compile on FreeBSD

# syslog-ng


## Version of syslog-ng

syslog-ng 3.16.1 release

## Platform

FreeBSD 11.1


# Issue

```
/bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I.  -I./lib -I./modules -I./lib -I./modules -I./tests/loggen -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -pthread  -I./lib/eventlog/src -I./lib/eventlog/src -I/usr/local/include  -I/usr/include   -isystem ./lib/ivykis/src/include -isystem ./lib/ivykis/src/include -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros  -O2 -g -pthread -Wno-initializer-overrides -O2 -pipe -fstack-protector -fno-strict-aliasing -MT tests/loggen/tests_loggen_libloggen_helper_la-loggen_helper.lo -MD -MP -MF tests/loggen/.deps/tests_loggen_libloggen_helper_la-loggen_helper.Tpo -c -o tests/loggen/tests_loggen_libloggen_helper_la-loggen_helper.lo `test -f 'tests/loggen/loggen_helper.c' || echo './'`tests/loggen/loggen_helper.c
libtool: compile:  cc -DHAVE_CONFIG_H -I. -I./lib -I./modules -I./lib -I./modules -I./tests/loggen -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -pthread -I./lib/eventlog/src -I./lib/eventlog/src -I/usr/local/include -I/usr/include -isystem ./lib/ivykis/src/include -isystem ./lib/ivykis/src/include -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -O2 -g -pthread -Wno-initializer-overrides -O2 -pipe -fstack-protector -fno-strict-aliasing -MT tests/loggen/tests_loggen_libloggen_helper_la-loggen_helper.lo -MD -MP -MF tests/loggen/.deps/tests_loggen_libloggen_helper_la-loggen_helper.Tpo -c tests/loggen/loggen_helper.c  -fPIC -DPIC -o tests/loggen/.libs/tests_loggen_libloggen_helper_la-loggen_helper.o
tests/loggen/loggen_helper.c:47:14: warning: implicit declaration of function
      'socket' is invalid in C99 [-Wimplicit-function-declaration]
  int sock = socket(dest_addr->sa_family, sock_type, 0);
             ^
tests/loggen/loggen_helper.c:47:30: error: incomplete definition of type
      'struct sockaddr'
  int sock = socket(dest_addr->sa_family, sock_type, 0);
                    ~~~~~~~~~^
/usr/include/netdb.h:140:9: note: forward declaration of 'struct sockaddr'
        struct  sockaddr *ai_addr;      /* binary address */
                ^
tests/loggen/loggen_helper.c:54:20: error: use of undeclared identifier
      'SOCK_STREAM'
  if (sock_type == SOCK_STREAM)
                   ^
tests/loggen/loggen_helper.c:63:7: warning: implicit declaration of function
      'connect' is invalid in C99 [-Wimplicit-function-declaration]
  if (connect(sock, dest_addr, dest_addr_len) < 0)
      ^
tests/loggen/loggen_helper.c:90:32: error: use of undeclared identifier
      'AF_INET6'
  hints.ai_family = use_ipv6 ? AF_INET6 : AF_INET;
                               ^
tests/loggen/loggen_helper.c:90:43: error: use of undeclared identifier
      'AF_INET'
  hints.ai_family = use_ipv6 ? AF_INET6 : AF_INET;
                                          ^
tests/loggen/loggen_helper.c:144:21: error: use of undeclared identifier
      'AF_UNIX'
  saun.sun_family = AF_UNIX;
                    ^
2 warnings and 5 errors generated.
gmake[4]: *** [Makefile:15111: tests/loggen/tests_loggen_libloggen_helper_la-loggen_helper.lo] Error 1
gmake[3]: *** [Makefile:17933: all-recursive] Error 1
gmake[2]: *** [Makefile:7334: all] Error 2
gmake[2]: Leaving directory '/root/syslog-ng316/work/syslog-ng-3.16.1'
*** Error code 1
```

After adding also to tests/loggen/ssl_plugin/ssl_plugin.c the loggen part now compiles.

Next problem (or should this go into a separate issue?):

```
/bin/sh ./libtool  --tag=CC   --mode=link cc -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros  -O2 -g -pthread -Wno-initializer-overrides -O2 -pipe -fstack-protector -fno-strict-aliasing  -fstack-protector -o syslog-ng/syslog-ng syslog-ng/main.o -L./lib/.libs -lsyslog-ng   -lrt -L/usr/local/lib -Wl,--export-dynamic -lgmodule-2.0 -lgthread-2.0 -pthread -lglib-2.0 -lintl  ./lib/eventlog/src/libevtlog.la   -L/usr/local/lib -lpcre    
libtool: link: cc -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -O2 -g -pthread -Wno-initializer-overrides -O2 -pipe -fstack-protector -fno-strict-aliasing -fstack-protector -o syslog-ng/.libs/syslog-ng syslog-ng/main.o -Wl,--export-dynamic -pthread  -L./lib/.libs /root/syslog-ng316/work/syslog-ng-3.16.1/lib/.libs/libsyslog-ng.so -lrt -L/usr/local/lib -lgmodule-2.0 -lgthread-2.0 -lglib-2.0 -lintl ./lib/eventlog/src/.libs/libevtlog.so -lpcre -pthread -Wl,-rpath -Wl,/usr/local/lib
/root/syslog-ng316/work/syslog-ng-3.16.1/lib/.libs/libsyslog-ng.so: undefined reference to `secret_storage_wipe'
cc: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[4]: *** [Makefile:10786: syslog-ng/syslog-ng] Error 1
gmake[3]: *** [Makefile:17933: all-recursive] Error 1
gmake[2]: *** [Makefile:7334: all] Error 2
gmake[2]: Leaving directory '/root/syslog-ng316/work/syslog-ng-3.16.1'
*** Error code 1
```

--- a/tests/loggen/file_reader.h
+++ b/tests/loggen/file_reader.h
@@ -21,8 +21,8 @@
  *
  */
 
-#ifndef __FILE_READER_H__
-#define __FILE_READER_H__
+#ifndef LOGGEN_FILE_READER_H_INCLUDED
+#define LOGGEN_FILE_READER_H_INCLUDED
 
 #include "compat/glib.h"
 #include <glib.h>

--- a/tests/loggen/loggen.c
+++ b/tests/loggen/loggen.c
@@ -247,7 +247,7 @@ init_logline_generator(GPtrArray *plugin_array)
   int framing;
   if (syslog_proto && !noframing)
     framing = 1;
-  else if(!syslog_proto && require_framing && !noframing)
+  else if (!syslog_proto && require_framing && !noframing)
     framing = 1;
   else
     framing = 0;
@@ -265,7 +265,7 @@ init_csv_statistics(void)
   /* message counter for csv output */
   thread_stat_count = (gint64 *) g_malloc0(global_plugin_option.active_connections * sizeof(gint64));
   thread_stat_count_last = (gint64 *) g_malloc0(global_plugin_option.active_connections * sizeof(gint64));
-  if(csv)
+  if (csv)
     {
       /* print CSV header and initial line about time zero */
       printf("ThreadId;Time;Rate;Count\n");

--- a/tests/loggen/loggen.c
+++ b/tests/loggen/loggen.c
@@ -29,8 +29,10 @@
 #include "logline_generator.h"
 
 #include <stdio.h>
+#include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <signal.h>
 #include <string.h>
 #include <gmodule.h>
 #include <errno.h>

--- a/tests/loggen/loggen_helper.c
+++ b/tests/loggen/loggen_helper.c
@@ -22,9 +22,15 @@
  */
 
 #include "loggen_helper.h"
+
 #include <syslog-ng-config.h>
+#include <string.h>
 #include <unistd.h>
+#include <time.h>
+#include <arpa/inet.h>
 #include <sys/time.h>
+#include <sys/types.h>
+#include <sys/socket.h>
 #include <openssl/err.h>
 
 static int debug = 0;

--- a/tests/loggen/loggen_helper.h
+++ b/tests/loggen/loggen_helper.h
@@ -21,8 +21,8 @@
  *
  */
 
-#ifndef __HELPER_H__
-#define __HELPER_H__
+#ifndef LOGGEN_HELPER_H_INCLUDED
+#define LOGGEN_HELPER_H_INCLUDED
 
 #include "compat/compat.h"
 #include "compat/glib.h"

--- a/tests/loggen/loggen_helper.h
+++ b/tests/loggen/loggen_helper.h
@@ -36,6 +36,10 @@
 
 #include <libgen.h>
 
+#define MAX_MESSAGE_LENGTH 8192
+#define USEC_PER_SEC      1000000
+#define CONNECTION_TIMEOUT_SEC 5
+
 int get_debug_level(void);
 void set_debug_level(int new_debug);
 unsigned long time_val_diff_in_usec(struct timeval *t1, struct timeval *t2);
@@ -48,21 +52,17 @@ int connect_unix_domain_socket(int sock_type, const char *path);
 SSL *open_ssl_connection(int sock_fd);
 void close_ssl_connection(SSL *ssl);
 
-#define MAX_MESSAGE_LENGTH 8192
-#define USEC_PER_SEC      1000000
-#define CONNECTION_TIMEOUT_SEC 5
-
-#define ERROR(format,...) do{\
-  fprintf(stderr,"error [%s:%s:%d] ",basename(__FILE__),__func__,__LINE__);\
-  fprintf(stderr,format,##__VA_ARGS__);\
-}while(0)
+#define ERROR(format,...) do {\
+  fprintf(stderr, "error [%s:%s:%d] ", basename(__FILE__), __func__, __LINE__);\
+  fprintf(stderr, format, ##__VA_ARGS__);\
+} while (0)
 
 /* debug messages can be turned on by "--debug" command line option */
-#define DEBUG(format,...) do{\
+#define DEBUG(format,...) do {\
   if (!get_debug_level()) \
     break; \
-  fprintf(stdout,"debug [%s:%s:%d] ",basename(__FILE__),__func__,__LINE__); \
-  fprintf(stdout,format,##__VA_ARGS__);\
-}while(0)
+  fprintf(stdout, "debug [%s:%s:%d] ", basename(__FILE__), __func__, __LINE__); \
+  fprintf(stdout, format, ##__VA_ARGS__);\
+} while(0)
 
 #endif

--- a/tests/loggen/loggen_helper.h
+++ b/tests/loggen/loggen_helper.h
@@ -27,6 +27,8 @@
 #include "compat/compat.h"
 #include "compat/glib.h"
 
+#include <stddef.h>
+#include <stdio.h>
 #include <glib.h>
 #include <sys/un.h>
 #include <netdb.h>

--- a/tests/loggen/loggen_plugin.h
+++ b/tests/loggen/loggen_plugin.h
@@ -21,8 +21,8 @@
  *
  */
 
-#ifndef __LOGGEN_PLUGIN_H__
-#define __LOGGEN_PLUGIN_H__
+#ifndef LOGGEN_PLUGIN_H_INCLUDED
+#define LOGGEN_PLUGIN_H_INCLUDED
 
 #include "compat/glib.h"
 #include <gmodule.h>

--- a/tests/loggen/loggen_plugin.h
+++ b/tests/loggen/loggen_plugin.h
@@ -28,6 +28,8 @@
 #include <gmodule.h>
 #include <sys/time.h>
 
+#define LOGGEN_PLUGIN_INFO "loggen_plugin_info"
+
 typedef struct _plugin_option
 {
   int message_length;
@@ -59,7 +61,6 @@ typedef int (*generate_message_func)(char *buffer, int buffer_size, int thread_i
 typedef void (*set_generate_message_func)(generate_message_func gen_message);
 typedef int (*get_thread_count_func)(void);
 typedef gboolean (*is_plugin_activated_func)(void);
-#define LOGGEN_PLUGIN_INFO "loggen_plugin_info"
 
 typedef struct _plugin_info
 {

--- a/tests/loggen/logline_generator.c
+++ b/tests/loggen/logline_generator.c
@@ -21,15 +21,16 @@
  *
  */
 
-#include <stdio.h>
-#include <sys/time.h>
-#include <sys/types.h>
-#include <unistd.h>
-#include <string.h>
-#include <glib.h>
-
 #include "logline_generator.h"
 #include "loggen_helper.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <glib.h>
 
 static char line_buf_template[MAX_MESSAGE_LENGTH + 1];
 static int pos_timestamp1 = 0;

--- a/tests/loggen/logline_generator.h
+++ b/tests/loggen/logline_generator.h
@@ -21,8 +21,8 @@
  *
  */
 
-#ifndef __LOGLINE_GENERATOR__
-#define __LOGLINE_GENERATOR__
+#ifndef LOGLINE_GENERATOR_H_INCLUDED
+#define LOGLINE_GENERATOR_H_INCLUDED
 
 int generate_log_line(char *buffer, int buffer_length, int syslog_proto, int thread_id, unsigned long seq);
 int prepare_log_line_template(int syslog_proto, int framing, int message_length, char *sdata_value);

--- a/tests/loggen/socket_plugin/socket_plugin.c
+++ b/tests/loggen/socket_plugin/socket_plugin.c
@@ -21,11 +21,15 @@
  *
  */
 
+#include "compat/glib.h"
 #include "loggen_plugin.h"
 #include "loggen_helper.h"
 
+#include <time.h>
 #include <signal.h>
+#include <stddef.h>
 #include <stdio.h>
+#include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <string.h>

--- a/tests/loggen/ssl_plugin/ssl_plugin.c
+++ b/tests/loggen/ssl_plugin/ssl_plugin.c
@@ -21,16 +21,19 @@
  *
  */
 
+#include "compat/glib.h"
 #include "loggen_plugin.h"
 #include "loggen_helper.h"
+#include "crypto.h"
 
+#include <stddef.h>
 #include <stdio.h>
+#include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <string.h>
 #include <netdb.h>
 
-#include <crypto.h>
 #include <openssl/crypto.h>
 #include <openssl/x509.h>
 #include <openssl/pem.h>

